File: double_loop/TriangleSearch/step2.rb
# 配列 A の要素数 N とその要素 A_i (1 ≦ i ≦ N) が与えられるので、A についてのかけ算表 B を出力してください。かけ算表は N * N の二次元配列の形式とし、B の i 行 j 列の要素 B_ij について、B_ij = Ai * Aj (1 ≦ i , j ≦ N) が成り立つものとします。

# 例として、A = [1,2,3] のとき B は

# 1 2 3
# 2 4 6
# 3 6 9

# となります。

n = gets.to_i

array = gets.split(" ").map(&:to_i)

ans = []
n.times do |i|
    temp = []
    n.times do |j|
       temp <<  array[i] * array[j]
    end
    ans << temp.join(" ")
end

puts ans

File: Python/STEINS;GATE/TabooSearchForExceptionHandling/step2.py
# 長さ n の 2 つの数列 A = {a_1, a_2, ..., a_n}, B = {b_1, b_2, ..., b_n} が与えられます。
# 数列の差 C = {b_1-a_1, b_2-a_2, ..., b_n-a_n} を求めてください。


n = int(input())
nums = list(map(int,input().split()))
nums2 = list(map(int,input().split()))

ans = []
for i in range(n):
    ans.append(str(nums2[i] - nums[i]))
print(" ".join(ans))


# 解答2
n = int(input())
a = [int(x) for x in input().split()]
b = [int(x) for x in input().split()]

for i in range(n):
    if i < n - 1:
        print(b[i] - a[i], end=" ")
    else:
        print(b[i] - a[i])